From d3b243d06400033937d2c0bf15c60afcc0c1e3e2 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Wed, 6 May 2026 05:17:22 -0700
Subject: [PATCH] Force-enable all settings

---
 frontend/libretro_core_options.h | 32 --------------------------------
 1 file changed, 32 deletions(-)

diff --git a/frontend/libretro_core_options.h b/frontend/libretro_core_options.h
index e049c6f4..123490da 100644
--- a/frontend/libretro_core_options.h
+++ b/frontend/libretro_core_options.h
@@ -61,27 +61,21 @@ struct retro_core_option_v2_category option_cats_us[] = {
       "Video",
       "Configure base display parameters."
    },
-#ifdef GPU_NEON
    {
       "gpu_neon",
       "GPU Plugin",
       "Configure low-level settings of the NEON GPU plugin."
    },
-#endif
-#ifdef GPU_PEOPS
    {
       "gpu_peops",
       "GPU Plugin (Advanced)",
       "Configure low-level settings of the P.E.Op.S. GPU plugin."
    },
-#endif
-#ifdef GPU_UNAI
    {
       "gpu_unai",
       "GPU Plugin (Advanced)",
       "Configure low-level settings of the UNAI GPU plugin."
    },
-#endif
    {
       "audio",
       "Audio",
@@ -164,19 +158,14 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#if defined(HAVE_CDROM) || defined(USE_ASYNC_CDROM)
 #define V(x) { #x, NULL }
    {
       "pcsx_rearmed_cd_readahead",
       "CD read-ahead",
       NULL,
       "Reads the specified amount of sectors ahead of time to try to avoid later stalls. "
-#ifdef HAVE_CDROM
       "Affects both physical CD-ROM and CD images. "
-#endif
-#if !defined(_3DS) && !defined(VITA)
       "333000 will try to read the complete disk (requires an additional 750MB of RAM)."
-#endif
       ,
       NULL,
       "system",
@@ -184,16 +173,12 @@ struct retro_core_option_v2_definition option_defs_us[] = {
          V(0),  V(1),  V(2),  V(3),  V(4),  V(5),  V(6),  V(7),
          V(8),  V(9),  V(10), V(11), V(12), V(13), V(14), V(15),
          V(16), V(32), V(64), V(128), V(256), V(512), V(1024),
-#if !defined(_3DS) && !defined(VITA)
 	 V(333000),
-#endif
          { NULL, NULL},
       },
       "12",
    },
 #undef V
-#endif
-#ifndef DRC_DISABLE
    {
       "pcsx_rearmed_drc",
       "Dynamic Recompiler",
@@ -208,7 +193,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#if !defined(LIGHTREC) && defined(NDRC_THREAD)
    {
       "pcsx_rearmed_drc_thread",
       "DynaRec threading",
@@ -224,8 +208,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "auto",
    },
-#endif
-#endif // DRC_DISABLE
    {
       "pcsx_rearmed_psxclock",
       "PSX CPU Clock Speed (%)",
@@ -329,7 +311,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "enabled",
 #endif
    },
-#ifdef USE_ASYNC_GPU
    {
       "pcsx_rearmed_gpu_thread_rendering",
       "Threaded Rendering",
@@ -345,7 +326,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "auto",
    },
-#endif
    {
       "pcsx_rearmed_frameskip_type",
       "Frameskip",
@@ -581,7 +561,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "0",
    },
 #undef V
-#ifdef GPU_NEON
    {
       "pcsx_rearmed_neon_interlace_enable_v2",
       "(GPU) Show Interlaced Video",
@@ -639,8 +618,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#endif /* GPU_NEON */
-#ifdef GPU_PEOPS
    {
       "pcsx_rearmed_show_gpu_peops_settings",
       "Show Advanced P.E.Op.S. GPU Settings",
@@ -767,8 +744,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* GPU_PEOPS */
-#ifdef GPU_UNAI
    {
       "pcsx_rearmed_show_gpu_unai_settings",
       "Show Advanced UNAI GPU Settings",
@@ -783,7 +758,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#ifndef GPU_UNAI_NO_OLD
    {
       "pcsx_rearmed_gpu_unai_old_renderer",
       "(GPU) Old renderer",
@@ -798,7 +772,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif
    {
       "pcsx_rearmed_gpu_unai_blending",
       "(GPU) Texture Blending",
@@ -855,7 +828,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* GPU_UNAI */
    {
       "pcsx_rearmed_spu_reverb",
       "Audio Reverb Effects",
@@ -922,7 +894,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#ifdef USE_ASYNC_SPU
    {
       "pcsx_rearmed_spu_thread",
       "Threaded SPU",
@@ -937,7 +908,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif
    {
       "pcsx_rearmed_show_input_settings",
       "Show Input Settings",
@@ -1703,7 +1673,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#if !defined(DRC_DISABLE) && !defined(LIGHTREC)
    {
       "pcsx_rearmed_nocompathacks",
       "Disable Automatic Compatibility Hacks",
@@ -1760,7 +1729,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* !DRC_DISABLE && !LIGHTREC */
    {
       "pcsx_rearmed_nostalls",
       "Disable CPU/GTE Stalls",
-- 
2.50.1 (Apple Git-155)

