From 0740b938129304d88cde237efacbcea480d7003a Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Wed, 6 May 2026 05:17:22 -0700
Subject: [PATCH] Force-enable all settings

---
 frontend/libretro_core_options.h | 30 ------------------------------
 1 file changed, 30 deletions(-)

diff --git a/frontend/libretro_core_options.h b/frontend/libretro_core_options.h
index bf2ee3c6..2bf3819b 100644
--- a/frontend/libretro_core_options.h
+++ b/frontend/libretro_core_options.h
@@ -61,27 +61,21 @@ struct retro_core_option_v2_category option_cats_us[] = {
       "Video",
       "Configure base display parameters."
    },
-#ifdef GPU_NEON
    {
       "gpu_neon",
       "GPU Plugin",
       "Configure low-level settings of the NEON GPU plugin."
    },
-#endif
-#ifdef GPU_PEOPS
    {
       "gpu_peops",
       "GPU Plugin (Advanced)",
       "Configure low-level settings of the P.E.Op.S. GPU plugin."
    },
-#endif
-#ifdef GPU_UNAI
    {
       "gpu_unai",
       "GPU Plugin (Advanced)",
       "Configure low-level settings of the UNAI GPU plugin."
    },
-#endif
    {
       "audio",
       "Audio",
@@ -181,19 +175,14 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "shared",
    },
-#if defined(HAVE_CDROM) || defined(USE_ASYNC_CDROM)
 #define V(x) { #x, NULL }
    {
       "pcsx_rearmed_cd_readahead",
       "CD read-ahead",
       NULL,
       "Reads the specified amount of sectors ahead of time to try to avoid later stalls. "
-#ifdef HAVE_CDROM
       "Affects both physical CD-ROM and CD images. "
-#endif
-#if !defined(_3DS) && !defined(VITA)
       "333000 will try to read the complete disk (requires an additional 750MB of RAM)."
-#endif
       ,
       NULL,
       "system",
@@ -209,8 +198,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "12",
    },
 #undef V
-#endif
-#ifndef DRC_DISABLE
    {
       "pcsx_rearmed_drc",
       "Dynamic Recompiler",
@@ -225,7 +212,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#if !defined(LIGHTREC) && defined(NDRC_THREAD)
    {
       "pcsx_rearmed_drc_thread",
       "DynaRec threading",
@@ -241,8 +227,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "auto",
    },
-#endif
-#endif // DRC_DISABLE
    {
       "pcsx_rearmed_psxclock",
       "PSX CPU Clock Speed (%)",
@@ -346,7 +330,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "enabled",
 #endif
    },
-#ifdef USE_ASYNC_GPU
    {
       "pcsx_rearmed_gpu_thread_rendering",
       "Threaded Rendering",
@@ -362,7 +345,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "auto",
    },
-#endif
    {
       "pcsx_rearmed_frameskip_type",
       "Frameskip",
@@ -598,7 +580,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "0",
    },
 #undef V
-#ifdef GPU_NEON
    {
       "pcsx_rearmed_neon_interlace_enable_v2",
       "(GPU) Show Interlaced Video",
@@ -656,8 +637,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#endif /* GPU_NEON */
-#ifdef GPU_PEOPS
    {
       "pcsx_rearmed_show_gpu_peops_settings",
       "Show Advanced P.E.Op.S. GPU Settings",
@@ -784,8 +763,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* GPU_PEOPS */
-#ifdef GPU_UNAI
    {
       "pcsx_rearmed_show_gpu_unai_settings",
       "Show Advanced UNAI GPU Settings",
@@ -800,7 +777,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#ifndef GPU_UNAI_NO_OLD
    {
       "pcsx_rearmed_gpu_unai_old_renderer",
       "(GPU) Old renderer",
@@ -815,7 +791,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif
    {
       "pcsx_rearmed_gpu_unai_blending",
       "(GPU) Texture Blending",
@@ -872,7 +847,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* GPU_UNAI */
    {
       "pcsx_rearmed_spu_reverb",
       "Audio Reverb Effects",
@@ -939,7 +913,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#ifdef USE_ASYNC_SPU
    {
       "pcsx_rearmed_spu_thread",
       "Threaded SPU",
@@ -954,7 +927,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif
    {
       "pcsx_rearmed_show_input_settings",
       "Show Input Settings",
@@ -1720,7 +1692,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#if !defined(DRC_DISABLE) && !defined(LIGHTREC)
    {
       "pcsx_rearmed_nocompathacks",
       "Disable Automatic Compatibility Hacks",
@@ -1777,7 +1748,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* !DRC_DISABLE && !LIGHTREC */
    {
       "pcsx_rearmed_nostalls",
       "Disable CPU/GTE Stalls",
-- 
2.50.1 (Apple Git-155)

