From e93f09e5fd8c2a064215e66e44598f4ec58ff832 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Thu, 20 May 2021 16:11:18 -0700
Subject: [PATCH] Force-enable NEON and Dynarec settings

---
 frontend/libretro_core_options.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/frontend/libretro_core_options.h b/frontend/libretro_core_options.h
index 813e30a..d316c25 100644
--- a/frontend/libretro_core_options.h
+++ b/frontend/libretro_core_options.h
@@ -479,7 +479,7 @@ struct retro_core_option_definition option_defs_us[] = {
 #endif
    },
 
-#if defined(LIGHTREC) || defined(NEW_DYNAREC)
+#if true
    {
       "pcsx_rearmed_drc",
       "Dynamic Recompiler",
@@ -493,7 +493,7 @@ struct retro_core_option_definition option_defs_us[] = {
    },
 #endif /* LIGHTREC || NEW_DYNAREC */
 
-#ifdef NEW_DYNAREC
+#if true
    {
       "pcsx_rearmed_psxclock",
       "PSX CPU Clock",
@@ -584,7 +584,7 @@ struct retro_core_option_definition option_defs_us[] = {
    },
 #endif /* NEW_DYNAREC */
 
-#ifdef GPU_NEON
+#if true
    {
       "pcsx_rearmed_neon_interlace_enable",
       "Enable Interlacing Mode",
@@ -969,7 +969,7 @@ struct retro_core_option_definition option_defs_us[] = {
       "disabled",
    },
 
-#ifdef NEW_DYNAREC
+#if true
    {
       "pcsx_rearmed_nosmccheck",
       "(Speed Hack) Disable SMC Checks",
-- 
2.30.2

