From d2dc905e2bb4498ea77476156a3213b16b15d557 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Tue, 5 Dec 2023 14:37:09 -0800
Subject: [PATCH] Force-enable all settings

---
 frontend/libretro_core_options.h | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/frontend/libretro_core_options.h b/frontend/libretro_core_options.h
index a0cd2ac1..f3f042fa 100644
--- a/frontend/libretro_core_options.h
+++ b/frontend/libretro_core_options.h
@@ -61,27 +61,21 @@ struct retro_core_option_v2_category option_cats_us[] = {
       "Video",
       "Configure base display parameters."
    },
-#ifdef GPU_NEON
    {
       "gpu_neon",
       "GPU Plugin",
       "Configure low-level settings of the NEON GPU plugin."
    },
-#endif
-#ifdef GPU_PEOPS
    {
       "gpu_peops",
       "GPU Plugin (Advanced)",
       "Configure low-level settings of the P.E.Op.S. GPU plugin."
    },
-#endif
-#ifdef GPU_UNAI
    {
       "gpu_unai",
       "GPU Plugin (Advanced)",
       "Configure low-level settings of the UNAI GPU plugin."
    },
-#endif
    {
       "audio",
       "Audio",
@@ -97,13 +91,11 @@ struct retro_core_option_v2_category option_cats_us[] = {
       "Compatibility Fixes",
       "Configure settings/workarounds required for correct operation of specific games."
    },
-#if !defined(DRC_DISABLE) && !defined(LIGHTREC)
    {
       "speed_hack",
       "Speed Hacks (Advanced)",
       "Configure hacks that may improve performance at the expense of decreased accuracy/stability."
    },
-#endif
    { NULL, NULL, NULL },
 };
 
@@ -182,7 +174,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "sync",
    },
 #endif
-#ifndef DRC_DISABLE
    {
       "pcsx_rearmed_drc",
       "Dynamic Recompiler",
@@ -197,7 +188,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#endif
    {
       "pcsx_rearmed_psxclock",
       "PSX CPU Clock Speed",
@@ -309,7 +299,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "enabled",
 #endif
    },
-#ifdef THREAD_RENDERING
    {
       "pcsx_rearmed_gpu_thread_rendering",
       "Threaded Rendering",
@@ -325,7 +314,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif
    {
       "pcsx_rearmed_frameskip_type",
       "Frameskip",
@@ -466,7 +454,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "0",
    },
 #undef V
-#ifdef GPU_NEON
    {
       "pcsx_rearmed_neon_interlace_enable_v2",
       "(GPU) Show Interlaced Video",
@@ -510,8 +497,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* GPU_NEON */
-#ifdef GPU_PEOPS
    {
       "pcsx_rearmed_show_gpu_peops_settings",
       "Show Advanced P.E.Op.S. GPU Settings",
@@ -638,8 +623,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* GPU_PEOPS */
-#ifdef GPU_UNAI
    {
       "pcsx_rearmed_show_gpu_unai_settings",
       "Show Advanced UNAI GPU Settings",
@@ -714,7 +697,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "disabled",
 #endif
    },
-#endif /* GPU_UNAI */
    {
       "pcsx_rearmed_spu_reverb",
       "Audio Reverb Effects",
@@ -781,7 +763,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#if P_HAVE_PTHREAD
    {
       "pcsx_rearmed_spu_thread",
       "Threaded SPU",
@@ -796,7 +777,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif // P_HAVE_PTHREAD
    {
       "pcsx_rearmed_show_input_settings",
       "Show Input Settings",
@@ -1524,7 +1504,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#if !defined(DRC_DISABLE) && !defined(LIGHTREC)
    {
       "pcsx_rearmed_nocompathacks",
       "Disable Automatic Compatibility Hacks",
@@ -1581,7 +1560,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* !DRC_DISABLE && !LIGHTREC */
    {
       "pcsx_rearmed_nostalls",
       "Disable CPU/GTE Stalls",
-- 
2.37.0.windows.1

