From 713422c8e17407fa708c03ff3327c736b5daf097 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Sat, 14 Dec 2024 19:33:09 -0800
Subject: [PATCH] Force-enable all settings

---
 frontend/libretro_core_options.h | 32 --------------------------------
 1 file changed, 32 deletions(-)

diff --git a/frontend/libretro_core_options.h b/frontend/libretro_core_options.h
index a07491b6..03bdb69b 100644
--- a/frontend/libretro_core_options.h
+++ b/frontend/libretro_core_options.h
@@ -61,27 +61,21 @@ struct retro_core_option_v2_category option_cats_us[] = {
       "Video",
       "Configure base display parameters."
    },
-#ifdef GPU_NEON
    {
       "gpu_neon",
       "GPU Plugin",
       "Configure low-level settings of the NEON GPU plugin."
    },
-#endif
-#ifdef GPU_PEOPS
    {
       "gpu_peops",
       "GPU Plugin (Advanced)",
       "Configure low-level settings of the P.E.Op.S. GPU plugin."
    },
-#endif
-#ifdef GPU_UNAI
    {
       "gpu_unai",
       "GPU Plugin (Advanced)",
       "Configure low-level settings of the UNAI GPU plugin."
    },
-#endif
    {
       "audio",
       "Audio",
@@ -163,19 +157,14 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#if defined(HAVE_CDROM) || defined(USE_ASYNC_CDROM)
 #define V(x) { #x, NULL }
    {
       "pcsx_rearmed_cd_readahead",
       "CD read-ahead",
       NULL,
       "Reads the specified amount of sectors ahead of time to try to avoid later stalls. "
-#ifdef HAVE_CDROM
       "Affects both physical CD-ROM and CD images. "
-#endif
-#if !defined(_3DS) && !defined(VITA)
       "333000 will try to read the complete disk (requires an additional 750MB of RAM)."
-#endif
       ,
       NULL,
       "system",
@@ -183,16 +172,12 @@ struct retro_core_option_v2_definition option_defs_us[] = {
          V(0),  V(1),  V(2),  V(3),  V(4),  V(5),  V(6),  V(7),
          V(8),  V(9),  V(10), V(11), V(12), V(13), V(14), V(15),
          V(16), V(32), V(64), V(128), V(256), V(512), V(1024),
-#if !defined(_3DS) && !defined(VITA)
 	 V(333000),
-#endif
          { NULL, NULL},
       },
       "12",
    },
 #undef V
-#endif
-#ifndef DRC_DISABLE
    {
       "pcsx_rearmed_drc",
       "Dynamic Recompiler",
@@ -207,7 +192,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#if !defined(LIGHTREC) && defined(NDRC_THREAD)
    {
       "pcsx_rearmed_drc_thread",
       "DynaRec threading",
@@ -223,8 +207,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "auto",
    },
-#endif
-#endif // DRC_DISABLE
    {
       "pcsx_rearmed_psxclock",
       "PSX CPU Clock Speed (%)",
@@ -334,7 +316,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "enabled",
 #endif
    },
-#ifdef THREAD_RENDERING
    {
       "pcsx_rearmed_gpu_thread_rendering",
       "Threaded Rendering",
@@ -350,7 +331,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif
    {
       "pcsx_rearmed_frameskip_type",
       "Frameskip",
@@ -522,7 +502,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "0",
    },
 #undef V
-#ifdef GPU_NEON
    {
       "pcsx_rearmed_neon_interlace_enable_v2",
       "(GPU) Show Interlaced Video",
@@ -580,8 +559,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#endif /* GPU_NEON */
-#ifdef GPU_PEOPS
    {
       "pcsx_rearmed_show_gpu_peops_settings",
       "Show Advanced P.E.Op.S. GPU Settings",
@@ -708,8 +685,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* GPU_PEOPS */
-#ifdef GPU_UNAI
    {
       "pcsx_rearmed_show_gpu_unai_settings",
       "Show Advanced UNAI GPU Settings",
@@ -724,7 +699,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#ifndef GPU_UNAI_NO_OLD
    {
       "pcsx_rearmed_gpu_unai_old_renderer",
       "(GPU) Old renderer",
@@ -739,7 +713,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif
    {
       "pcsx_rearmed_gpu_unai_blending",
       "(GPU) Texture Blending",
@@ -800,7 +773,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       "disabled",
 #endif
    },
-#endif /* GPU_UNAI */
    {
       "pcsx_rearmed_spu_reverb",
       "Audio Reverb Effects",
@@ -867,7 +839,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "enabled",
    },
-#if P_HAVE_PTHREAD
    {
       "pcsx_rearmed_spu_thread",
       "Threaded SPU",
@@ -882,7 +853,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif // P_HAVE_PTHREAD
    {
       "pcsx_rearmed_show_input_settings",
       "Show Input Settings",
@@ -1648,7 +1618,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#if !defined(DRC_DISABLE) && !defined(LIGHTREC)
    {
       "pcsx_rearmed_nocompathacks",
       "Disable Automatic Compatibility Hacks",
@@ -1705,7 +1674,6 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       },
       "disabled",
    },
-#endif /* !DRC_DISABLE && !LIGHTREC */
    {
       "pcsx_rearmed_nostalls",
       "Disable CPU/GTE Stalls",
-- 
2.43.0

